### ISSUE TITLE
[FEATURE] Setting to disable the Accuracy & Difficulty prompt window, or move accuracy/difficulty to buttons in chat roll message

### ISSUE THREAD

**Issue by ChromeWisp** [Eranziel/foundryvtt-lancer#155]
**Is your feature request related to a problem? Please describe.**
Not a huge problem, but I am bothered that this window comes up for every roll (in case accuracy or difficulty apply) and must always be closed with an extra click or keypress.

**Describe the solution you'd like**
A setting to disable these Accuracy and Difficulty window prompts appearing when attempting a roll (so that players/GMs can instead use a macro to roll any acc/diff d6s themselves, for example).

**Describe alternatives you've considered**
Instead of popping up an accuracy & difficulty window when rolling, have the roll happen immediately _without_ any accuracy/difficulty, and then in the roll's chat message show buttons to add accuracies or difficulties to the roll's current results! Would look similar to the D&D 5e chat messages for item uses, where there are buttons for attack rolls and damage. Actually that would be _awesome!_

**Additional context**
Some mockup images of the accuracy & difficulty buttons being on the roll's chat message instead...

![Normal](https://user-images.githubusercontent.com/4585257/115977309-5b1a4080-a53c-11eb-8b5b-0662d93dc7b0.png)
![Diff](https://user-images.githubusercontent.com/4585257/115977313-69685c80-a53c-11eb-868a-d0d9efaab4ee.png)
![Acc1](https://user-images.githubusercontent.com/4585257/115977314-6b322000-a53c-11eb-9809-c14d9119b19b.png)
![Acc2](https://user-images.githubusercontent.com/4585257/115977315-6c634d00-a53c-11eb-8768-e32dad8263d6.png)

In this concept, below the results for a roll, there's a difficulty button and an accuracy button, with a number displayed between them to show the current modifier from any accuracy/difficulty (by default, +0d6). Pressing these buttons adds or subtracts accuracy or difficulty dice respectively in the roll's results. So in these screenshots, player character Vostok might add on a difficulty and have his rolled 11 visibly change to a 9. Or maybe instead he has two accuracy so he presses the accuracy button twice and the text changes to show his roll bumped up to a 16.

**Discord contact info**
If you are willing to answer questions regarding this feature request on Discord, or if you want to be notified when it is fixed, please leave your Discord ID here.
e.g. Wisp#8374

**Comment by Grygon** [comment]
I'll grab this as a potential thing, but probably not until v1.1--very cool idea though, I love it

**Comment by VexxMyst** [comment]
I'm not sure on how foundry handles its log, but I feel like a roll can't be modified after being made.
However, I like the large buttons handling both accuracy and difficulty. I'd back this for a popup revamp.

**Comment by ChromeWisp** [comment]
I see your point... Well what about this: instead of the accuracy & difficulty buttons rolling new d6s on the spot...have it so that when you make the initial d20 roll, a series of d6s gets rolled in the background but not added to your roll right away. Then, when you press the accuracy button, it adds in that first pre-rolled d6 to your roll. Then if you press the accuracy button again, you instead get the best of the first _two_ d6s in the series added to your roll. If you try and press the difficulty button and then accuracy button again to "save scum" and hopefully get a better result, it won't happen because it's not re-rolling the d6s every time, it's just choosing how many of the d6s in the series to pick the highest out of.

![Roll](https://user-images.githubusercontent.com/4585257/116008124-979b7a00-a5d8-11eb-9b8c-b39201604c49.png)
![Roll1Diff](https://user-images.githubusercontent.com/4585257/116008129-9c602e00-a5d8-11eb-9a18-0434840c753f.png)
![Roll1Acc](https://user-images.githubusercontent.com/4585257/116008138-a4b86900-a5d8-11eb-9e6e-810d92baa798.png)
![Roll2Acc](https://user-images.githubusercontent.com/4585257/116008140-a71ac300-a5d8-11eb-898f-4b3aa59a91f2.png)

**Comment by VexxMyst** [comment]
That's a neat idea on how to do it. If it can be done, then yeah, I like that.

**Comment by Grygon** [comment]
Yeah I like that--and tbh, we don't even really need to have separate acc/diff d6s

**Comment by Phenomen** [comment]
I prefer a BetterRolls/MidiQOL method: Hold Shift for Adv or Ctrl for Dis roll. It would save a lot of space on the chat card.

**Comment by msprijatelj** [comment]
Haven't commented on this repo before, but interested in following this system's development! 

Chiming in as a GM here: Either having something like ChromeWisp suggested that edits rolls in post, or alternatively having a small window that could clearly increase/decrease accuracy + other modifiers **_before_** a roll is made would be immensely helpful and would reduce "verification fatigue".  Example of the latter from Fantasy Grounds Unity, for 5e:
![image](https://user-images.githubusercontent.com/44978587/128616295-937d8c78-974a-43ba-b517-131599d3e057.png)
Where the buttons seen can be enabled/disabled to modify the next roll made.  Then **_even if_** a player or myself forgets a to apply Hard Cover or something, we can just retry the roll with "Hard Cover" enabled.  Make the cost of making a mistake cheaper (i.e. fewer clicks) so rerolls/retries are less painful, yeah?
